### PR TITLE
2367: Swagger migration to springdoc-openapi lib

### PR DESCRIPTION
**Pull request by RudzMaksim**
**Description:**
-migrated Swagger documentation to springdoc-openapi lib to fix issues related to required fields.

### CODE REVIEW

File: src/main/kotlin/com/epam/brn/config/SwaggerConfig.kt [hunk 1]
```diff
@@ -1,46 +1,65 @@
...
+    @Bean
+    fun rolesAllowedCustomizer(): OperationCustomizer? {
+        return OperationCustomizer { operation, handlerMethod ->
+            var allowedRoles: Array<String>? = null
+            var rolesAllowedAnnotation = handlerMethod.getMethodAnnotation(RolesAllowed::class.java)
+            if (rolesAllowedAnnotation != null) {
+                allowedRoles = rolesAllowedAnnotation.value
+            } else {
+                rolesAllowedAnnotation = handlerMethod.method.declaringClass.getAnnotation(RolesAllowed::class.java)
+                if (rolesAllowedAnnotation != null) {
```
Comment (ElenaSpb) [inline src/main/kotlin/com/epam/brn/config/SwaggerConfig.kt:37]: we don't use {} in single conditions
https://github.com/Brain-up/brn/wiki/Coding-Standards

File: src/main/kotlin/com/epam/brn/config/SwaggerConfig.kt [hunk 2]
```diff
@@ -1,46 +1,65 @@
...
+            }
 
-    private fun apiInfo(): ApiInfo {
-        return ApiInfoBuilder()
-            .title("Brain up project")
-            .description("REST API for brn")
-            .contact(Contact("Elena.Moshnikova", "https://www.epam.com/", "brainupproject@yandex.ru"))
-            .build()
+            val sb = StringBuilder("Roles: ")
+            if (allowedRoles != null) {
```
Comment (ElenaSpb) [inline src/main/kotlin/com/epam/brn/config/SwaggerConfig.kt:43]: same, please

File: src/main/kotlin/com/epam/brn/config/SwaggerConfig.kt [hunk 3]
```diff
@@ -1,46 +1,65 @@
...
-    protected fun addResourceHandlers(registry: ResourceHandlerRegistry) {
-        registry.addResourceHandler("swagger-ui.html")
-            .addResourceLocations("classpath:/META-INF/resources/")
-        registry.addResourceHandler("/webjars/**")
-            .addResourceLocations("classpath:/META-INF/resources/webjars/")
+    @Bean
+    fun sortTagsCustomiser(): OpenApiCustomiser {
+        return OpenApiCustomiser { openApi ->
+            openApi.tags.sortBy { it.name }
+        }
```
Comment (ElenaSpb) [inline src/main/kotlin/com/epam/brn/config/SwaggerConfig.kt:63]: why don't you want to make it more Kotlinish

@Bean
    fun sortTagsCustomiser(): OpenApiCustomiser =
         OpenApiCustomiser { openApi ->
            openApi.tags.sortBy { it.name }
        }

File: src/main/kotlin/com/epam/brn/config/SwaggerConfig.kt (hunk 1 above)
Comment (RudzMaksim) [reply src/main/kotlin/com/epam/brn/config/SwaggerConfig.kt]: Fixed all comments

**Review by ElenaSpb — APPROVED** [review]

**Pull request merged**